### ISSUE TITLE
[splici Reference] Mixed R2 read length

### ISSUE THREAD

**Issue by TobiTekath** [COMBINE-lab/alevin-fry#21]
Hi there,
I am very excited about your tutorial about using a splici reference with alevin-fry and would love to test it out.
I have one question about the read_length parameter that is used to define the flank length added to each intron:

In my single-cell data, I have some samples with 90bp R2 reads, and some with 150bp R2 reads. As far as I understood, it should be safe to go for a read_length of 90bp, right? Or do you advise something different?

Thanks a lot.

Best,
Tobi

**Comment by rob-p** [comment]
Hi @TobiTekath,

  Thanks for trying out the splici index and USA mode quantification!  We'd be very interested to hear your experience after you've had a chance to run it.
  
  I'm going to ping @mohsenzakeri here as well, as he has done some testing regarding the effect of flank length choices in single-cell data.  However, I'll say that, as long as you plan to follow the tutorial in adding the `S` and `A` sub-matrices to get the final counts, I think that pretty much any choice of flank length between 90 and 150 should give very similar results.  Let me explain my thinking a bit below.

  * Setting a longer flank length : The effect of a longer flank length is that more sequence flanking the introns will be included in the "intron" sequence.  This will generally lead to some more reads being marked as "ambiguous" rather than strictly "spliced".  However, by adding the `S` and `A` matrices, you'll recover these reads anyway.  
  
  * Setting a shorter flank length : On the other hand, if the flank length is set smaller than the length of some reads (say as 90), then some longer reads that span the intron-exon boundary will properly map to neither.  That is, if they only slightly extend into the intron, then they will extend past the end of the flank on the intronic sequence (and so won't map there) and the intronic part of the sequence won't be accounted for by the spliced sequence (and so they won't map there).  _However_, since, in a standard single-cell experiment, you are looking for the `S`+`A` counts anyway, and these fragments would have been regarded as `U`, failing to map them should not present a problem.  Further, I suspect there will be relatively few such fragments anyway, as they have to fall in a quite particular way at the intron-exon boundary for this to happen.

In short, for processing single-cell data and extracting the total "spliced" counts at the end, then any flank length in the range of your R2 lengths should perform well.  @mohsenzakeri can share some data from an experiment we performed varying the flank length on simulated data to see the effect.

Please let us know if you have any other questions, and we'd be happy to have your feedback once you've had a chance to quantify your data.

Thanks!
Rob

**Comment by TobiTekath** [comment]
Hi @rob-p,

thank you very much for this quick but nonetheless extensive answer. Your outlined thoughts are extremely helpful for me.

Just as a note, that I will then go for longer flanking regions, as I am also interested in the U counts for velocity estimation. And as long as I add up S and A counts, that should not hurt.

Thanks again.

**Comment by mohsenzakeri** [comment]
Hi @TobiTekath ,

As @rob-p mentioned the accuracy of the quantification with the splici index is pretty robust with respect to the flank length one chooses for building the splici index. You can take a look at the following results which is evaluated based on different flank lengths. In the following table, the accuracy of quantification of a simulated single cell RNAseq sample is displayed based on different metrics: 

- spearman correlation
- mard_drop_nan: mean of relative differences after removing the entries which are zero both in the truth and the estimation
- fp: the mean of number of genes in each cell which have count zero in the truth but are estimated to have a non-zero count
- fn: the mean of number of genes in each cell which have non-zero count in the truth but are estimated to be unexpressed

<html>
<body>


flank-length | spearman | mard_drop_nan | fp | fn
-- | -- | -- | -- | --
85.0 | 0.988318 | 0.026321 | 0.010969 | 0.011832
90.0 | 0.988337 | 0.026280 | 0.010935 | 0.011830
95.0 | 0.988346 | 0.026261 | 0.010916 | 0.011830
100.0 | 0.988355 | 0.026242 | 0.010897 | 0.011832
105.0 | 0.988367 | 0.026216 | 0.010872 | 0.011832
110.0 | 0.988375 | 0.026199 | 0.010854 | 0.011834
115.0 | 0.988381 | 0.026189 | 0.010842 | 0.011836
120.0 | 0.988389 | 0.026172 | 0.010825 | 0.011836
125.0 | 0.988404 | 0.026144 | 0.010794 | 0.011837
130.0 | 0.988437 | 0.026081 | 0.010733 | 0.011836
135.0 | 0.988456 | 0.026042 | 0.010694 | 0.011838
140.0 | 0.988463 | 0.026029 | 0.010679 | 0.011838
145.0 | 0.988472 | 0.026010 | 0.010660 | 0.011839


</body>
</html>

I hope you find this auxiliary information helpful about the effect of the flank length on the quality of the estimations provided by alevin-fry. Feel free to let us know if you have any further questions.

Best,
Mohsen

**Comment by TobiTekath** [comment]
Hi @mohsenzakeri ,

thank you very much for sharing this data - the approach indeed looks very stable. Now I am quite confident I will get meaningful results, regardless of the chosen read-length.

Best,
Tobi
